src/Dialect/ONNX/ONNXOps.cpp

```diff
@@ -106,6 +106,24 @@ LogicalResult shapeHelperInferMultipleShapes(OP *op, Value typeOper) {
   return success();
 }
 
+// Handle shape inference for binary operators.
+template <class OP, class ADAPTOR>
+static LogicalResult inferShapeForBinaryOps(
+    OP *op, Type elementType = nullptr) {
+  ADAPTOR operandAdaptor(*op);
+  for (const auto &operand : operandAdaptor.getOperands())
+    if (!hasShapeAndRank(operand))
+      return success();
+
+  auto resultTy = op->getOperand(0).getType().template cast<ShapedType>();
+  for (unsigned i = 1; i < op->getNumOperands(); ++i) {
+    auto nextTy = op->getOperand(i).getType().template cast<ShapedType>();
```

Comment:
Avoid `auto` here and the one at Line 118.

Reply:
I used `auto` here because the type is clear given that on the RHS there is a cast to `ShapedType`.